projects/igniteui-angular/src/lib/icon/types.ts

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+import { Theme } from 'igniteui-webcomponents/theming/types.js';
+
+// Exported internal types
+export type IconThemeKey = Theme | 'default';
+
+export type IconReferencePair = {
+  alias: IconMeta;
+  target: IconMeta;
+  overwrite: boolean;
+};
+export type IconType = "svg" | "font" | "liga";
+
+export type IconReference = IconMeta & FamilyMeta;
+
+export type MetaReference = {
+  alias: IconMeta;
+  target: Map<IconThemeKey, IconMeta>;
+};
+
+export interface FamilyMeta {
```

Comment:
And I think this one should also be moved in the public API part too since it's used in the service `setFamily` method.
And with it `IconType` apparently.

Reply:
And `IconReference` because of `getIconRef`, basically all the old exports from `icon.service.ts` should be exported from the public API